homeassistant/components/assist_pipeline/pipeline.py

```diff
@@ -394,6 +401,13 @@ def __post_init__(self) -> None:
             )
         pipeline_data.pipeline_runs[self.pipeline.id][self.id] = PipelineRunDebug()
 
+        # Directory to save audio for each pipeline run.
+        # Configured in YAML for assist_pipeline.
+        if debug_recording_dir := self.hass.data[DATA_CONFIG].get(
+            "debug_recording_dir"
+        ):
+            self.debug_recording_dir = Path(debug_recording_dir) / self.id
```

Comment:
Given that many devices can use the same pipeline, should we consider storing it in a device directory instead if that is available ?

Reply:
I've reworked it to use the pipeline name and device id when available. So the directory structure will either be:

`<debug_recording_dir>/<pipeline.name>/<run.id>`
or
`<debug_recording_dir>/<device_id>/<pipeline.name>/<run.id>`